### PR TITLE
Closing readers based on errors

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

File: filebeat/harvester/log_file.go [hunk 1]
```diff
@@ -157,7 +139,13 @@ func (r *LogFile) errorChecks(err error) error {
 
 func (r *LogFile) wait() {
 	// Wait before trying to read file wr.ch reached EOF again
-	time.Sleep(r.backoff)
+	timer := time.NewTimer(r.backoff)
+
+	select {
+	case <-r.done:
+		return
+	case <-timer.C:
```
Comment (urso) [inline filebeat/harvester/log_file.go]: one can use `<-time.After(r.backoff)` here

Comment (ruflin) [inline filebeat/harvester/log_file.go]: changed

File: filebeat/harvester/log_file.go [hunk 2]
```diff
@@ -156,8 +138,12 @@ func (r *LogFile) errorChecks(err error) error {
...
 
 func (r *LogFile) wait() {
-	// Wait before trying to read file wr.ch reached EOF again
-	time.Sleep(r.backoff)
+	// Wait before trying to read file again. File reached EOF.
+	select {
+	case <-r.done:
+		return
+	case <-time.After(r.backoff):
+	}
```
Comment (urso) [inline filebeat/harvester/log_file.go]: where is `wait` being used? We need to distinguish be return value if done channel was closed or wait was completed?

Comment (ruflin) [inline filebeat/harvester/log_file.go]: It does not matter, as the first thing after `wait()` returns is that it checks again if the channel was closed.

**Pull request merged**